### PR TITLE
[https://nvbugs/6550099][fix] Raise no-top-k equivalence tolerance

### PR DESCRIPTION
**Pull request by lori-ren**
## Dev Engineer Review

- Updated `test_no_topk_matches_full` to account for FP32 variation at the nucleus cutoff.
- Raised the no-top-k equivalence tolerance from `1e-5` to `1e-4`.
- Removed the matching waiver from `tests/integration/test_lists/waives.txt`.
- The change is scoped to the affected test. No API or public entity changes were made.
- The waiver path matches the modified test. No duplicate or format issues are evident.

## QA Engineer Review

- Modified test: `test_no_topk_matches_full`.
- The test is not listed in `test-db/` or `qa/`.
- The waiver entry was removed.
- The full test file and 30 consecutive runs of the affected case passed on an A100.
- Verdict: needs follow-up because CBTS coverage data is unavailable.

## Description

Root cause: the assertion's tolerance was tighter than the precision of the operator it compares. FULL and NO_TOPK are bit-identical up to the top-p renorm, so nothing in the sampling path differs. At the nucleus cutoff fp32 precision decides whether one token is kept, and that token is worth up to 7.8e-5 of mass in input batch — about 8x the assertion's atol=1e-5. The test asked for more than fp32 offers, so it failed whenever that boundary token moved.

Fixed by raising atol to 1e-4.
Validated on A100: full file passes, 30 consecutive runs of the case pass. The waiver is removed in the fix PR.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Comment by lori-ren** [comment]
/bot run --stage-list "DGX_H100-PyTorch-4" --disable-fail-fast

**Comment by lori-ren** [comment]
/bot run --stage-list "DGX_H100-PyTorch-4" --disable-fail-fast

File: tests/unittest/_torch/speculative/hw_agnostic/test_advanced_sampling_mode.py [hunk 1]
```diff
@@ -85,9 +85,9 @@ def test_resolve_advanced_sampling_filters(mode, expect_top_k_none, expect_top_p
 def test_no_topk_matches_full(top_p_val):
     """With top_k disabled, NO_TOPK skips the top_k mask kernel (a no-op at k=vocab)
     and yields the same sampling distribution as FULL. We compare the resulting
-    probability distributions rather than the sampled tokens: the flashinfer top_k mask
-    at k=vocab injects ~1e-8 fp noise that leaves the distribution unchanged but can flip
-    an individual sampled token across GPU archs, so exact-token equality is not portable.
+    probability distributions rather than the sampled tokens, and with a tolerance above
+    the top-p renorm's fp32 precision: at the nucleus cutoff that precision decides
+    whether one token is kept, worth up to ~8e-5 of mass for this batch.
```
Comment (zhaoyangwang-nvidia) [inline tests/unittest/_torch/speculative/hw_agnostic/test_advanced_sampling_mode.py]: The old docstring named the source of the divergence (the `k=vocab` `top_k_mask_logits` injecting ~1e-8 of fp noise); this rewrite drops it, and the PR description states FULL and NO_TOPK are "bit-identical up to the top-p renorm". If the inputs to `top_p_renorm_probs` were truly identical, the same deterministic kernel would make the same cutoff decision and this test could not fail intermittently. Could you keep both halves of the chain here (mask-kernel noise -> boundary token flips at the nucleus cutoff -> ~8e-5 of mass), so a future reader knows what the tolerance is actually sized against?

File: tests/unittest/_torch/speculative/hw_agnostic/test_advanced_sampling_mode.py [hunk 2]
```diff
@@ -105,7 +105,7 @@ def test_no_topk_matches_full(top_p_val):
     )
     probs_full = su.compute_probs_from_logits(logits.clone(), temperatures, ek_full, ep_full)
     probs_no_topk = su.compute_probs_from_logits(logits.clone(), temperatures, ek_nt, ep_nt)
-    assert torch.allclose(probs_full, probs_no_topk, atol=1e-5, rtol=0)
+    assert torch.allclose(probs_full, probs_no_topk, atol=1e-4, rtol=0)
```
Comment (zhaoyangwang-nvidia) [inline tests/unittest/_torch/speculative/hw_agnostic/test_advanced_sampling_mode.py]: I reproduced the top-p cutoff mechanics for this input shape, and two things stand out. First, the quantity `allclose` sees is the post-renorm delta (~8.1e-5 here, about 11% above the 7.8e-5 boundary-token mass the atol was sized against, because dropping the boundary token rescales the ~1900 kept tokens), so `1e-4` leaves only ~1.2x headroom. Second, that delta is a max-over-rows order statistic with no structural bound: re-running the same input distribution under other seeds gives 8.3e-5, 9.0e-5 and 4.5e-4, i.e. one in five would exceed `1e-4` by 4.5x. The tolerance therefore holds because the seed is pinned, not because it is above fp32 precision, and any change to seed/batch/vocab or to the flashinfer softmax can reopen this bug. Since the property under test is "NO_TOPK is not a real filter" (a real one moves O(0.1) of mass), could we assert something bounded by construction instead - e.g. `(probs_full - probs_no_topk).abs().sum(-1).max() < 1e-3`, or derive the atol from the measured boundary-token mass?

File: tests/unittest/_torch/speculative/hw_agnostic/test_advanced_sampling_mode.py (hunk 1 above)
Comment (lori-ren) [reply tests/unittest/_torch/speculative/hw_agnostic/test_advanced_sampling_mode.py]: Done. Now both halves of causal chain is marked in the comment.

File: tests/unittest/_torch/speculative/hw_agnostic/test_advanced_sampling_mode.py (hunk 2 above)
Comment (lori-ren) [reply tests/unittest/_torch/speculative/hw_agnostic/test_advanced_sampling_mode.py]: Agreed, we now changed to the bounded-by-construction form you suggested.

**Comment by lori-ren** [comment]
/bot run --stage-list "DGX_H100-PyTorch-4" --disable-fail-fast
